substrate/client/authority-discovery/src/worker/tests.rs

```diff
@@ -611,15 +674,240 @@ fn strict_accept_address_with_peer_signature() {
 		tester.remote_authority_public.into(),
 		&tester.remote_key_store,
 		Some(&TestSigner { keypair: &tester.remote_node_key }),
+		true,
+	);
+
+	let cached_remote_addresses = tester.process_value_found(true, kv_pairs).0;
+
+	assert_eq!(
+		Some(HashSet::from([addr])),
+		cached_remote_addresses,
+		"Expect worker to only cache `Multiaddr`s with `PeerId`s.",
+	);
+}
+
+#[test]
+fn strict_accept_address_without_creation_time() {
+	let mut tester = DhtValueFoundTester::new();
+	let addr = tester.multiaddr_with_peer_id(1);
+	let kv_pairs = build_dht_event(
+		vec![addr.clone()],
+		tester.remote_authority_public.into(),
+		&tester.remote_key_store,
+		Some(&TestSigner { keypair: &tester.remote_node_key }),
+		false,
+	);
+
+	let cached_remote_addresses = tester.process_value_found(true, kv_pairs).0;
+
+	assert_eq!(
+		Some(HashSet::from([addr])),
+		cached_remote_addresses,
+		"Expect worker to only cache `Multiaddr`s with `PeerId`s.",
+	);
+}
+
+#[test]
+fn keep_last_received_if_no_creation_time() {
+	let mut tester: DhtValueFoundTester = DhtValueFoundTester::new();
+	let addr = tester.multiaddr_with_peer_id(1);
+	let kv_pairs = build_dht_event(
+		vec![addr.clone()],
+		tester.remote_authority_public.into(),
+		&tester.remote_key_store,
+		Some(&TestSigner { keypair: &tester.remote_node_key }),
+		false,
+	);
+
+	let (cached_remote_addresses, network) = tester.process_value_found(true, kv_pairs);
+
+	assert_eq!(
+		Some(HashSet::from([addr])),
+		cached_remote_addresses,
+		"Expect worker to only cache `Multiaddr`s with `PeerId`s.",
```

Comment:
This text description seems copied across the tests. Is it relevant in all the places?

Reply:
Updated with proper messages